flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java

```diff
@@ -2844,4 +2845,35 @@ public void testSharedBufferClearing() throws Exception {
 			Mockito.verify(accessor, Mockito.times(1)).advanceTime(2);
 		}
 	}
+
+	/**
+	 * Test that can access the value of the previous stage directly in notFollowedBy.
+	 *
+	 * @see <a href="https://issues.apache.org/jira/browse/FLINK-15964">FLINK-15964</a>
+	 * @throws Exception
+	 */
+	@Test
+	public void testAccessPreviousStageInNotFollowedBy() throws Exception {
```

Comment:
What about move this test case to NotPatternITCase which is dedicated for not pattern?

Reply:
OK